### PR TITLE
Generate and upload SBOM

### PR DESCRIPTION
**Pull request by dragomirp**
## Issue
Generate SBOM manifests for future use by security auditing

## Solution
Generate SBOM and upload as artefact in publish workflows

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -29,9 +29,24 @@ jobs:
...
       - uses: actions/download-artifact@v3
         with:
           name: postgresql-rock
+      - name: Create SBOM
+        run: syft charmed-postgresql_${{env.tag}}_amd64.rock -o spdx-json=charmed-postgresql_${{env.tag}}_amd64.rock.spdx.json
+      - name: Upload SBOM
+        uses: actions/upload-artifact@v3
+        with:
+          path: "charmed-postgresql_${{env.tag}}_amd64.rock.spdx.json"
+          name: charmed-postgresql_${{env.tag}}_amd64.rock.spdx.json
```
Comment (dragomirp) [inline .github/workflows/publish.yaml]: There is an action that can do this, but there is a bug preventing usage with files https://github.com/anchore/sbom-action/issues/389

**Comment by dragomirp** [comment]
Tested with a personal repo.  The end result should look like [this](https://github.com/dragomirp/charmed-postgresql-rock/actions/runs/5290278775).

**Review by taurus-forever — COMMENTED** [review]
Unfortunately, we cannot merge it AS IS, as the workflow has access to GitHub secrets which will leak if https://raw.githubusercontent.com/anchore/syft/main/install.sh is compromised.

We have to extract it into the separate workflow without secrets and exchange SBOM via artifacts. Security... Security... Security...

File: .github/workflows/sbom.yaml
Comment (dragomirp) [inline .github/workflows/sbom.yaml:1]: Maybe we should merge this in the `build` workflow?

Comment (taurus-forever) [reply .github/workflows/sbom.yaml:1]: I would say, we will have to merge it as shared data-platform-workflow one day.
Meanwhile, I would prefer to keep it as separate workflow to indicate PoC availability here.

**Review by taurus-forever — APPROVED** [review]
LGTM, @carlcsaposs-canonical can you please comment it from your side. Tnx!

**Review by taurus-forever — APPROVED** [review]

**Review by marceloneppel — APPROVED** [review]

**Pull request merged**